### DISCUSSION TITLE
How to create a new window instance?

### DISCUSSION THREAD

**Discussion by qFlensT** [AllenDang/giu#459] [Q&A]
Hello everyone! My button's task is to **open a new window when it is clicked**. But when calling `NewMasterWindow().Run()` I get the **following error**: 
`"Expression: !Locked && "Cannot modify a locked ImFontAtlas between NewFrame() and EndFrame/Render()!""`

How can I solve it?

**My code:**

```
view.ASCConfigureMonitorBtn.OnClick(func() {
		filepath, err := dialog.File().Filter("Image File", "jpeg", "jpg", "png").Load()
		if err != nil {
			log.Error(err.Error())
			return
		}

		img, err := os.Open(filepath)
		defer img.Close()

		imageData, _, err := image.Decode(img)
		if err != nil {
			log.Error(err.Error())
			return
		}
		imageBounds := imageData.Bounds()
		imageSize := imageBounds.Size()
		
		wnd := g.NewMasterWindow("Edit monitor", imageSize.X/2, imageSize.Y/2+300, g.MasterWindowFlagsNotResizable)

		wnd.Run(func() {
			g.SingleWindow().Layout(
				g.ImageWithFile(filepath).Size(float32(imageSize.X/2), float32(imageSize.Y/2)),
			)
		})
	})
```

**Comment by gucio321** [comment]
well @qFlensT the error is caused by the fact that the windows uses the same contex variable
however, making multiple windows instance may be an interessting feature...
@AllenDang is it just fault of our global-context system or it is something within imgui/imgui-go itself?

**Comment by AllenDang** [comment]
@gucio321 It's related to imgui's context management. The `docking` branch of imgui is the right way to handle this case, to open multiple native window. I intent to migrate long times ago but it is not that easy, need to change lots of code for go wrapper, and I intend to wait for go 1.18 with generic, to refine giu completly and migrate to docking at same time.
